salt/cloud/clouds/vmware.py

```diff
@@ -2350,6 +2393,33 @@ def create(vm_):
             )
             clone_spec.customization = custom_spec
 
+        if customization and (devices and 'network' in list(devices.keys())) and 'Windows' in object_ref.config.guestFullName:
+            global_ip = vim.vm.customization.GlobalIPSettings()
+
+            if 'dns_servers' in list(vm_.keys()):
+                global_ip.dnsServerList = vm_['dns_servers']
+
+            identity = vim.vm.customization.Sysprep()
+            identity.guiUnattended = vim.vm.customization.GuiUnattended()
+            identity.guiUnattended.autoLogon = False
+            identity.guiUnattended.password = vim.vm.customization.Password()
+            identity.guiUnattended.password.value = vm_['win_password']
+            identity.guiUnattended.password.plainText = True
+            identity.userData = vim.vm.customization.UserData()
+            hostName = vm_name.split('.')[0]
+            identity.userData.fullName = hostName
+            identity.userData.orgName = "Organization-Name"
+            identity.userData.computerName = vim.vm.customization.FixedName()
+            identity.userData.computerName.name = domain
```

Comment:
Why is this set to ``domain``? This should be the computer hostname instead. According to the docs, 
```
The computer name of the (Windows) virtual machine. Computer name may contain letters (A-Z), numbers(0-9) and hyphens (-) but no spaces or periods (.). The name may not consists entirely of digits. On Vista computer name is restricted to 15 characters in length. If the computer name is longer than 15 characters, it will be truncated to 15 characters.
```
Please put appropriate checks in place to make sure the criteria is fulfilled.